crates/node/builder/src/exex.rs

```diff
@@ -16,10 +16,12 @@ trait LaunchExEx<Node: FullNodeComponents>: Send {
     ) -> impl Future<Output = eyre::Result<impl Future<Output = eyre::Result<()>> + Send>> + Send;
 }
 
-type BoxExEx = BoxFuture<'static, eyre::Result<()>>;
+/// A boxed exex future.
+pub type BoxExEx = BoxFuture<'static, eyre::Result<()>>;
 
 /// A version of [`LaunchExEx`] that returns a boxed future. Makes the trait object-safe.
-pub(crate) trait BoxedLaunchExEx<Node: FullNodeComponents>: Send {
+pub trait BoxedLaunchExEx<Node: FullNodeComponents>: Send {
```

Comment:
can we make these `pub(crate)`? i dont see any reason to impl this as an end user. as soon as we make it pub we need to treat it somewhat as stable api

Reply:
these must be public if we want to make the exex launcher reusable.

this is necessary if we want to write chain specific launch functions

Reply:
might be a dumb q, but why would you want to re-use the exex launcher outside of this context?

Reply:
in order to write custom launch impls,
for example OPLaucnher, EthLauncher, etc... which need to be implemented externally

Reply:
hmm i thought the idea was that you provide the types and can override eg engine impl + evm and that's about it?

Reply:
this is how it works now for OP and ETH which reuse the same launcher, this will get more complex once we need to write more custom launchers, for example with different consensus, or not use the engine API at all